compiler/src/iree/compiler/Codegen/Dialect/GPU/TargetUtils/ConfigUtils.cpp

```diff
@@ -78,4 +57,52 @@ SmallVector<int64_t> deriveThreadTileSizes(Operation *op) {
   return tileSizes;
 }
 
+SmallVector<int64_t> deriveLinalgOpThreadTileSizes(linalg::LinalgOp linalgOp,
+                                                   int64_t numThreads) {
+  if (!linalgOp.hasPureTensorSemantics()) {
+    return {};
+  }
+  // TODO: Support multi-result
+  if (linalgOp->getNumResults() != 1) {
+    return {};
+  }
+  SmallVector<int64_t> loopRanges = linalgOp.getStaticLoopRanges();
+  int64_t vectorSize = kPreferredCopyNumBits /
+                       getElementTypeOrSelf(linalgOp->getResultTypes()[0])
+                           .getIntOrFloatBitWidth();
+  return getThreadTileSizesFromLoopRanges(loopRanges, numThreads, vectorSize);
+}
+
+SmallVector<int64_t>
+deriveIm2colOpThreadTileSizes(IREE::LinalgExt::Im2colOp im2colOp,
+                              int64_t numThreads) {
+  if (!im2colOp.hasPureTensorSemantics()) {
+    return {};
+  }
+  SmallVector<int64_t> loopRanges(im2colOp.getOutputType().getShape());
```

Comment:
side note: IMO, it's better to add `getStaticIterationBound` to TilingInterface or create a similar method in `Utils/`. Then we don't need to hardcode such logic here. I verified that it matches the `Im2colOp::getIterationDomain`, so it looks fine to me.

The concern is that we are duplicating such logics to other places, and I believe that this part should match the tiling implementation.

https://github.com/iree-org/iree/blob/45323df861d2386073ff0d3aa3fdebad14e164b2/compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/TilingInterfaceImpl.cpp#L1147-L1159

Reply:
Yeah, I agree. I will leave a TODO comment for now to replace once there is a more structured way of getting the static iteration bounds.